i3lock.c

```diff
@@ -684,9 +695,6 @@ int main(int argc, char *argv[]) {
         {NULL, no_argument, NULL, 0}
     };
 
-    if ((username = getenv("USER")) == NULL)
```

Comment:
It’s slightly cleaner to handle errors in the calling program instead of having a function which then terminates the program (see also: libraries must never call `exit()`).

In order to make this pull request smaller (which is always good), you could just call `getpwuid(geteuid())` in the main function and not introduce a new function at all.